File: java/src/main/java/com/example/hello/HelloController.java
package com.example.hello;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/hello")
public class HelloController {

    @GetMapping
    public String hello(@RequestParam(required = false) String name) {
        if (name == null || name.isBlank()) {
            return "Hello World!";
        }
        return "Hello " + name + "!";
    }
}

File: python/hello_controller.py
from fastapi import APIRouter


class HelloController:
    def __init__(self):
        self._router = APIRouter()
        self._router.add_api_route("/api/hello", self.hello, methods=["GET"])

    @property
    def router(self):
        return self._router

    def hello(self, name: str = None):
        if not name:
            return "Hello World!"
        return f"Hello {name}!"
